src/lava/proc/lif/models.py

```diff
@@ -145,9 +157,151 @@ def run_spk(self):
         updated_volt = decayed_volt + self.u + self.effective_bias
         self.v[:] = np.clip(updated_volt, neg_voltage_limit, pos_voltage_limit)
 
-        # Spike when exceeds threshold
-        # ----------------------------
-        s_out = self.v >= self.effective_vth
+    def reset_voltage(self, spike_vector: np.ndarray):
+        """Voltage reset behaviour. This can differ for different neuron
+            models.
+        """
+        self.v[spike_vector] = 0
+
+    def run_spk(self):
+        """The run function that performs the actual computation during
+        execution orchestrated by a PyLoihiProcessModel using the
+        LoihiProtocol
+        """
+        # Receive synaptic input
+        a_in_data = self.a_in.recv()
+
+        # Compute effective bias and threshold only once, not every time-step
+        if not self.isbiasscaled:
+            self.scale_bias()
+
+        if not self.isthrscaled:
+            self.scale_threshold()
+
+        self.subthr_dynamics(activation_in=a_in_data)
+
+        s_out = self.spiking_activation()
+
         # Reset voltage of spiked neurons to 0
-        self.v[s_out] = 0
+        self.reset_voltage(spike_vector=s_out)
         self.s_out.send(s_out)
+
+
+@implements(proc=LIF, protocol=LoihiProtocol)
+@requires(CPU)
+@tag('floating_pt')
+class PyLifModelFloat(AbstractPyLifModelFloat):
+    """Implementation of Leaky-Integrate-and-Fire neural process in floating
+    point precision. This short and simple ProcessModel can be used for quick
+    algorithmic prototyping, without engaging with the nuances of a fixed
+    point implementation.
+    """
+    s_out: PyOutPort = LavaPyType(PyOutPort.VEC_DENSE, bool, precision=1)
+    vth: float = LavaPyType(float, float)
+
+    def spiking_activation(self):
+        """Spiking activation function for LIF"""
+        return self.v >= self.vth
```

Comment:
L38 in process.py uses v>vth, and here v >= vth is used. The same criterion should be used across all LIF processes and models, otherwise bit-accuracy tests may fail.

Reply:
Fixed this. All spiking_activation() functions use strict inequality now.